packages/neptune-extension/src/common/components/checkout-modal/CheckoutModal.tsx

```diff
@@ -70,23 +71,37 @@ const CheckoutModal: React.FC<CheckoutModalProps> = ({
     [notebookId]
   );
 
-  async function handleSubmit() {
+  const dispatch = useDispatch();
+
+  async function checkoutNotebook() {
     if (checkpointId === undefined || notebookId === undefined || projectId === undefined) {
       return;
     }
 
-    setLoading(true);
-
     const notebook = await leaderboardClient.api.getNotebook({ id: notebookId });
     const content = await leaderboardClient.api.getCheckpointContent({ id: checkpointId });
 
     setDefaultProjectId(projectId);
 
     const newPath = await findNonExistantPath(notebook.path, platformNotebook);
     await platformNotebook.saveNotebookAndOpenInNewWindow(newPath, content);
+  }
 
-    setLoading(false);
-    onClose();
+  async function handleSubmit() {
+    setLoading(true);
+
+    try {
+      await checkoutNotebook();
+      onClose();
```

Comment:
UploadModal closes whether upload was successful or not.
CheckoutModal closes only on success.

Could we please unify the behaviour?